### DISCUSSION TITLE
Arquitetura Angular

### DISCUSSION THREAD

**Discussion by ipodjms** [frontendbr/forum#1324] [Perguntas e Respostas]
Galera, boa tarde!
Estou iniciando meu primeiro projeto em Angular. Sendo assim, estou com bastante dúvida de como montar a arquitetura. 

Como vocês fazem a arquitetura do Angular? Usam HOC? Tratam os services apenas para http e a logica fica num HOC? ou em outro service, ou mesmo service do http?

Se puderam compartilhar a arquitetura ou disponibilizar artigos que me ajudam a criar algo legal, iria ser muito massa!!

**Comment by rubenmarcus** [comment]
@ipodjms  
para leitura:
https://medium.com/@cyrilletuzi/architecture-in-angular-projects-242606567e40
https://netmedia.io/dev/angular-architecture-patterns-high-level-project-architecture_5589


Aqui na empresa eu montei a arquitetura, e junto com outro front mudamos conforme a necessidade. Ele tinha mais exp. em Angular e ajudou muito a remodelar.

hoje temos aqui:

modulo principal da aplicação
  - web ( todos modulos de web inseridos )
 - shared ( serviços ( sim sao pra http), helpers , pipes, decorators, models) 
 -  webview ( o app mobile consome )

dentro da web e da webview tambem tem os modulos shared, que são componentes reutilizados pelos submodulos desses modulos (ex: botões, banners,cards etc)


não usamos HOC... estou dando uma olhada aqui e tentando entender a vantagem de se usar no Angular
